src/ui/MLFacetManager/MLFacetManager.ts

```diff
@@ -0,0 +1,183 @@
+    }),
+    /**
+     * A function that defines the sort order for the facets.
+     *
+     * Called on every successful query reponse.
+     *
+     * The implementation of this method is identical to the Javascript `compareFunction` of the sort method for arrays.
+     * (see [Array.prototype.sort()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort))
+     *
+     * **Note:**
+     * > You cannot set this option directly in the component markup as an HTML attribute. You must either set it in the
+     * > [`init`]{@link init} call of your search interface (see
+     * > [Components - Passing Component Options in the init Call](https://developers.coveo.com/x/PoGfAQ#Components-PassingComponentOptionsintheinitCall)),
+     * > or before the `init` call, using the `options` top-level function (see
+     * > [Components - Passing Component Options Before the init Call](https://developers.coveo.com/x/PoGfAQ#Components-PassingComponentOptionsBeforetheinitCall)).
+     */
+    compareFacets: ComponentOptions.buildCustomOption<IMLFacetManagerCompareFacet>(() => {
+      return null;
+    })
+  };
+
+  // Children MLFacet components of the MLFacetManager
+  private mLFacets: MLFacet[];
+  private containerElement: HTMLElement;
+
+  /**
+   * Creates a new `MLFacetManager` instance.
+   *
+   * @param element The element from which to instantiate the component.
+   * @param options The component options.
+   * @param bindings The component bindings. Automatically resolved by default.
+   */
+  constructor(element: HTMLElement, public options?: IMLFacetManagerOptions, private bindings?: IComponentBindings) {
+    super(element, 'MLFacetManager');
+    this.options = ComponentOptions.initComponentOptions(element, MLFacetManager, options);
+
+    this.moveChildrenIntoContainer();
+    this.initEvents();
+  }
+
+  private resetContainer() {
+    this.containerElement && $$(this.containerElement).remove();
+    this.containerElement = $$('div', { className: 'coveo-ml-facet-manager-container' }).el;
+  }
+
+  private moveChildrenIntoContainer() {
+    this.resetContainer();
+    $$(this.element)
+      .children()
+      .forEach(child => this.containerElement.appendChild(child));
+    this.element.appendChild(this.containerElement);
+  }
+
+  private initEvents() {
+    this.bind.onRootElement(InitializationEvents.afterComponentsInitialization, () => this.handleAfterComponentsInitialization());
+    this.bind.onRootElement(QueryEvents.querySuccess, (data: IQuerySuccessEventArgs) => this.handleQuerySuccess(data));
+  }
+
+  private handleAfterComponentsInitialization() {
+    const allMLFacets = this.bindings.searchInterface.getComponents<MLFacet>('MLFacet');
+    this.mLFacets = allMLFacets.filter(mlFacet => this.element.contains(mlFacet.element));
+
+    if (!this.mLFacets.length) {
+      this.disable();
+    }
+  }
+
+  private handleQuerySuccess(data: IQuerySuccessEventArgs) {
+    if (this.disabled) {
```

Comment:
That's superfluous since you bound the event using `this.bind.onRootElement` , it automatically takes care of not calling the handler if the component is disabled.